### ISSUE TITLE
manual/javaGuide/main/sql/JavaEbean.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#568]
- https://github.com/garbagetown/playdocja/blob/2.2.0/documentation/2.2.0/manual/javaGuide/main/sql/JavaEbean.md

```
--- /Users/garbagetown/Desktop/2.1.5/manual/javaGuide/main/sql/JavaEbean.md 2013-09-19 22:53:02.000000000 +0900
+++ /Users/garbagetown/Desktop/2.2.0/manual/javaGuide/main/sql/JavaEbean.md 2013-09-19 10:11:22.000000000 +0900
@@ -2,7 +2,16 @@

 ## Configuring Ebean

-Play 2.0 comes with the [[Ebean| http://www.avaje.org/]] ORM. To enable it, add the following line to `conf/application.conf`:
+Play comes with the [Ebean](http://www.avaje.org/) ORM. To enable it, add javaEbean to your
+dependencies : 
+
+'''scala
+val appDependencies = Seq(
+  javaEbean
+)
+'''
+
+then add the following line to `conf/application.conf`:

 '''properties
 ebean.default="models.*"
@@ -39,11 +48,11 @@

 Note that Ebean will also make use of a `conf/orm.xml` file (if present), to configure `<entity-mappings>`.

-> For more information about Ebean, see the [[Ebean documentation | http://www.avaje.org/ebean/documentation.html]].
+> For more information about Ebean, see the [Ebean documentation](http://www.avaje.org/ebean/documentation.html).

 ## Using the play.db.ebean.Model superclass

-Play 2.0 defines a convenient superclass for your Ebean model classes. Here is a typical Ebean class, mapped in Play 2.0:
+Play defines a convenient superclass for your Ebean model classes. Here is a typical Ebean class, mapped in Play:

 '''java
 package models;
@@ -82,7 +91,7 @@

 > (1) Because Ebean class enhancement occurs *after* compilation, **do not expect Ebean-generated getter/setters to be available at compilation time.** If you'd prefer to code with them directly, either add the getter/setters explicitly yourself, or ensure that your model classes are compiled before the remainder of your project, eg. by putting them in a separate subproject.

-> (2) Enhancement of direct Ebean field access (enabling lazy loading) is only applied to Java classes, not to Scala. Thus, direct field access from Scala source files (including standard Play 2 templates) does not invoke lazy loading, often resulting in empty (unpopulated) entity fields. To ensure the fields get populated, either (a) manually create getter/setters and call them instead, or (b) ensure the entity is fully populated *before* accessing the fields.
+> (2) Enhancement of direct Ebean field access (enabling lazy loading) is only applied to Java classes, not to Scala. Thus, direct field access from Scala source files (including standard Play templates) does not invoke lazy loading, often resulting in empty (unpopulated) entity fields. To ensure the fields get populated, either (a) manually create getter/setters and call them instead, or (b) ensure the entity is fully populated *before* accessing the fields.

 As you can see, we've added a `find` static field, defining a `Finder` for an entity of type `Task` with a `Long` identifier. This helper field is then used to simplify querying our model:

@@ -101,18 +110,30 @@
     .ilike("name", "%coco%")
     .orderBy("dueDate asc")
     .findPagingList(25)
+    .setFetchAhead(false)
     .getPage(1);
 '''

 ## Transactional actions

-By default Ebean will not use transactions. However, you can use any transaction helper provided by Ebean to create a transaction. For example:
+By default Ebean will use transactions. However this transactions will be created before and commited or rollbacked after every single query, update, create or delete, as you can see here:
+
+'''
+//Created implicit transaction
+List<User> users =   
+            Ebean.find(User.class)  
+                .join("customer")  
+                .where().eq("state", UserState.ACTIVE)  
+                .findList();  
+//Transaction commited or rollbacked
+
+//Created implicit transaction
+Ebean.save(user);
+//Transaction commited or rollbacked

-:: Rob Bygrave - 
-The above statement is not correct. Ebean will use implicit transactions. To demarcate transactions you have 3 options: @Transactional, TxRunnable() or a beginTransaction(), commitTransaction()
+'''

-See http://www.avaje.org/ebean/introtrans.html for examples and an explanation.
-:: - end note
+So, if you want to do more than one action in the same transaction you can use TxRunnable and TxCallable:

 '''
 // run in Transactional scope...  
@@ -135,7 +156,7 @@
 });
 '''

-You can also annotate your action method with `@play.db.ebean.Transactional` to compose your action method with an `Action` that will automatically manage a transaction:
+You can also, if the class is a POJO one, annotate your action method with `@play.db.ebean.Transactional` to compose your action method with an `Action` that will automatically manage a transaction:

 '''
 @Transactional
@@ -144,4 +165,24 @@
 }
 '''

-> **Next:** [[Integrating with JPA | JavaJPA]]
\ No newline at end of file
+Or if you want a more traditional approach you can begin, commit and rollback transactions explicitly:
+
+'''
+Ebean.beginTransaction();  
+try {  
+    // fetch some stuff...  
+    User u = Ebean.find(User.class, 1);  
+    ...  
+  
+    // save or delete stuff...  
+    Ebean.save(u);  
+    ...  
+  
+    Ebean.commitTransaction();  
+      
+} finally {  
+    Ebean.endTransaction();  
+}  
+'''
+
+> **Next:** [[Integrating with JPA | JavaJPA]]

```

**Comment by garbagetown** [comment]
pull requested at #676

**Comment by wm3** [comment]
merged #676
